miri-script/src/main.rs

```diff
@@ -147,12 +150,30 @@ fn main() -> Result<()> {
         Some("build") => Command::Build { flags: args.collect() },
         Some("check") => Command::Check { flags: args.collect() },
         Some("test") => {
-            let bless = args.peek().is_some_and(|a| a.to_str() == Some("--bless"));
-            if bless {
-                // Consume the flag.
-                args.next().unwrap();
+            let mut target = std::env::var("MIRI_TEST_TARGET").ok();
+            let mut bless = false;
+
+            while let Some(arg) = args.peek().and_then(|s| s.to_str()) {
+                match arg {
+                    "--bless" => bless = true,
+                    "--target" => {
+                        // Skip "--target"
+                        args.next();
+                        target = args
+                            .peek()
+                            .map(|s| s.clone().into_string())
+                            .transpose()
+                            .map_err(|os_str| anyhow!("Invalid target string: {:?}", os_str))?;
```

Comment:
This looks unnecessarily complicated... if `args.peek()` is `None`, you can just throw an error.